sortinghat/parsing/grimoirelab.py

```diff
@@ -0,0 +1,358 @@
+            is_bot: false
+          github:
+            - vsekhark
+          email:
+            - vivek@****.com
+          enrollments:
+            - organization: Community
+              start: 1900-01-01
+              end: 2100-01-01
+
+        :parse json: YAML object to parse
+
+        :raise InvalidFormatError: raised when the format of the YAML is
+            not valid.
+        """
+        yaml = self.__load_yml(stream)
+
+        try:
+            for yid in yaml:
+
+                name = None
+                first_email = None
+                first_username = None
+                uuid = ''
+                is_bot = False
+
+                if "name" in yid["profile"]:
+                    name = yid["profile"]["name"]
+                if "is_bot" in yid['profile']:
+                    is_bot = yid['profile']['is_bot']
+
+                if 'email' in yid:
+                    first_email = self.__validate_email(yid['email'][0])
+
+                # I get there the first username available for the first
+                # Perceval backend found in the file
+                for pb in PERCEVAL_BACKENDS:
+                    if pb in yid:
+                        first_username = yid[pb][0]
+                        break
+
+                if name: uuid += name
+                if first_email: uuid += first_email
+                if first_username: uuid += first_username
+
+                #either first_email or first_username must exist
+                if (not first_email) and (not first_username):
+                    msg = "invalid GrimoireLab yaml format. At least " \
+                    "email or user account must be included"
+                    raise InvalidFormatError(cause=msg)
+
+                uid = UniqueIdentity(uuid=uuid)
+
+                identity = Identity(name=name, email=first_email,
+                                    username=first_username,
+                                    source=self.source, uuid=uuid)
+                uid.identities.append(identity)
+
+
+                # here we insert the name and username we've found with
+                # all the emails found in the file
+                if "email" in yid:
+                    for alt_email in yid["email"]:
+                        valid_email = self.__validate_email(alt_email)
+                        alt_email= self.__encode(valid_email)
+
+                        if alt_email == first_email:
+                            continue
+
+                        identity = Identity(name=name, email=alt_email,
+                                            username=first_username,
+                                            source=self.source, uuid=uuid)
+                        uid.identities.append(identity)
+
+
+                # now we insert all the usernames with the first name and
+                # email we found
+                first = True
+                for pb in PERCEVAL_BACKENDS:
+                    if pb in yid:
```

Comment:
Instead of adding a new level of indentation you can add a continue here checking that `pb` is not in `yid`.
```
if pb not in yid:
    continue
```